lib/Controller/FavoritesController.php

```diff
@@ -151,29 +252,51 @@ public function renameCategories($categories, $newName) {
     /**
      * @NoAdminRequired
      */
-    public function deleteFavorite($id) {
-        $favorite = $this->favoritesService->getFavoriteFromDB($id, $this->userId);
-        if ($favorite !== null) {
-            $this->favoritesService->deleteFavoriteFromDB($id);
-            return new DataResponse('DELETED');
+    public function deleteFavorite($id, $myMapId=null) {
+        if (is_null($myMapId) || $myMapId === "") {
+            $favorite = $this->favoritesService->getFavoriteFromDB($id, $this->userId);
+            if ($favorite !== null) {
+                $this->favoritesService->deleteFavoriteFromDB($id);
+                return new DataResponse('DELETED');
+            } else {
+                return new DataResponse('no such favorite', 400);
+            }
```

Comment:
```suggestion
            }
            return new DataResponse('no such favorite', 400);
```

Also this looks like it's using space instead of tabs here :) Might be a good idea to add php-cs-fix to this repo